perf-testing/test_cloudwatch.rb

```diff
@@ -0,0 +1,246 @@
+
+  separated_cbor_serde = separate_serde_data(thread[:cbor_serde_data])
+  write_test_output(test_case, 'CBOR', metrics, 'Serialization time (ms)', separated_cbor_serde[0], data)
+  write_test_output(test_case, 'CBOR', metrics, 'Deserialization time (ms)', separated_cbor_serde[1], data)
+  write_test_output(test_case, 'CBOR', metrics, 'Total request time (ms)', thread[:cbor_total_data], data)
+  write_test_output(test_case, 'CBOR', metrics, 'Request payload size (bytes)', separated_request_size[1], data)
+  write_test_output(test_case, 'CBOR', metrics, 'Response payload size (bytes)', separated_response_size[1], data)
+
+  clear_thread_data(thread)
+end
+
+thread = Thread.current
+thread[:query_serde_data] = []
+thread[:cbor_serde_data] = []
+thread[:query_total_data] = []
+thread[:cbor_total_data] = []
+thread[:request_size_data] = []
+thread[:response_size_data] = []
+thread[:warm] = false
+
+Aws::CloudWatch::Client.api.metadata['targetPrefix'] = 'GraniteServiceVersion20100801'
+Aws::CloudWatch::Client.remove_plugin(Aws::Plugins::Protocols::Query)
+query_cloudwatch = Aws::CloudWatch::Client.new(plugins: [Aws::Plugins::Protocols::Query], region: 'us-west-2')
+cbor_cloudwatch = Aws::CloudWatch::Client.new(plugins: [Aws::Plugins::Protocols::RpcV2], region: 'us-west-2')
+cbor_cloudwatch.config.api = cbor_cloudwatch.config.api.dup
+cbor_cloudwatch.config.api.metadata = cbor_cloudwatch.config.api.metadata.dup
+cbor_cloudwatch.config.api.metadata['protocol'] = 'smithy-rpc-v2-cbor'
+
+data = File.open('perf-testing/test-output/cloudwatch/data.txt', 'w')
+raw = File.open('perf-testing/test-output/cloudwatch/raw.txt', 'w')
+
+(0...WARMUP).each do
+  request = generate_put_metric_data_request(1, BASE_TIME, SUITE_ID)
+  query_cloudwatch.put_metric_data(request)
+  cbor_cloudwatch.put_metric_data(request)
+  request = generate_get_metric_data_request(1, BASE_TIME, SUITE_ID)
+  query_cloudwatch.get_metric_data(request)
+  cbor_cloudwatch.get_metric_data(request)
+end
+(0...WARMUP).each do
+  request = generate_list_metrics_request
+  query_cloudwatch.list_metrics(request)
+  cbor_cloudwatch.list_metrics(request)
+end
+thread[:warm] = true
+
+METRIC_COUNTS.each do |metrics|
+  (0...ITERATIONS).each do |i|
+    request = generate_put_metric_data_request(metrics, BASE_TIME, SUITE_ID)
+    t_query_total_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    query_cloudwatch.put_metric_data(request)
+    t_query_total_end = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    thread[:query_total_data] << format('%.3f', (t_query_total_end - t_query_total_start) * 1000.0).to_f
+
+    t_cbor_total_start = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    cbor_cloudwatch.put_metric_data(request)
+    t_cbor_total_end = Process.clock_gettime(Process::CLOCK_MONOTONIC)
+    thread[:cbor_total_data] << format('%.3f', (t_cbor_total_end - t_cbor_total_start) * 1000.0).to_f
+    sleep(2) if (i % 50).zero?
```

Comment:
Was this specified as part of the testing doc?

Edit: NVM - I see that it is - maybe add a comment for it.